### PR TITLE
IA refresh homepage focus

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/4837 by adding "focus" snippets (using snippets because this is data that will be updated infrequently, if ever)

admin views:
- homepage: https://foundation-s-ia-refresh-rnkrjt.herokuapp.com/cms/pages/3/edit/
- focus snippets: https://foundation-s-ia-refresh-rnkrjt.herokuapp.com/cms/snippets/wagtailpages/focusarea/

live view: https://foundation-s-ia-refresh-rnkrjt.herokuapp.com/en/

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/base.py [hunk 1]
```diff
@@ -438,6 +439,59 @@ class ParticipateHighlights2(ParticipateHighlightsBase):
...
+    url = models.URLField(
+        max_length=2048,
+        help_text='URL for this area of focus\' dedicated page.',
+    )
+
+    panels = [
+        ImageChooserPanel('interest_icon'),
+        FieldPanel('name'),
+        FieldPanel('description'),
+        FieldPanel('url'),
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/base.py:470]: @KalobTaulien do  you know if there is a way to make this use the same kind of URL dialog that we get in the RichText editor?

Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/pagemodels/base.py:470]: Yep, but it's a two pronged approach when using standard model fields. 

```
internal_url = models.ForeignKey('wagtailcore.Page', ..., blank=True)
external_url = models.URLField(..., blank=True) 

@property 
def url(self):
    if self.internal_url:
        return self.internal_url.url 
    return self.external_url 

panels = [
    ....
    PageChooserPanel("internal_url"),
    FieldPanel("external_url")
]
```

Note I didnt check that for typos, but that's the gist of it. We can also add page validation to make sure both fields are not set and help_text to help editors know what to do.

File: network-api/networkapi/wagtailpages/pagemodels/base.py [hunk 2]
```diff
@@ -438,6 +439,58 @@ class ParticipateHighlights2(ParticipateHighlightsBase):
     )
 
 
+@register_snippet
+class FocusArea(models.Model):
+    interest_icon = models.ForeignKey(
+        'wagtailimages.Image',
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/pagemodels/base.py:445]: Will this always be an image, never an svg?

File: network-api/networkapi/wagtailpages/pagemodels/base.py [hunk 3]
```diff
@@ -438,6 +439,58 @@ class ParticipateHighlights2(ParticipateHighlightsBase):
...
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        verbose_name = 'Area of focus'
+        verbose_name_plural = 'Areas of focus'
+
+
+class AreaOfFocus(WagtailOrderable, models.Model):
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/pagemodels/base.py:481]: I don't believe we need `models.Model` in here as `Orderable` is a `Model` already.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/areas_of_focus.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{% load wagtailimages_tags i18n %}
+
+<div class="row areas-of-focus">
+  <div class="col mb-4">
+    <h2 class="h4-heading">{% trans "Our areas of focus" %}</h4>
+    <div class="row">
+        {% with page.areas_of_focus.all as areas %}
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/areas_of_focus.html:7]: Points for using the `with` keyword!

File: network-api/networkapi/wagtailpages/pagemodels/base.py (hunk 2 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/pagemodels/base.py:445]: yeah, given the trouble with SVG, we only use static asset svg icons, but jpg/png works fine here, too, and is easier to update for staff.

File: network-api/networkapi/wagtailpages/pagemodels/base.py (hunk 3 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/pagemodels/base.py:481]: yep, good catch!

Comment (Pomax) [reply network-api/networkapi/wagtailpages/pagemodels/base.py:481]:
```suggestion
class AreaOfFocus(WagtailOrderable):
```

**Comment by Pomax** [comment]
soft blocking on https://github.com/mozilla/foundation.mozilla.org/issues/4936 for a few days

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/areas_of_focus.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+          {% with focus.area as area %}
+            <div class="col-12 col-md-4">
+              {% image area.interest_icon width-100 %}
+              <h2 class="h2-heading">{{ area.name }}</h2>
+              <p class="...">{{ area.description }}</p>
+              <a href="{{ area.url }}" class="...">{% trans "Learn more →" %}</a>
+            </div>
+          {% endwith %}
+        {% endfor %}
+      {% endwith %}
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/areas_of_focus.html:18]:
```suggestion
      {% for focus in page.areas_of_focus.all %}
        {% with focus.area as area %}
          <div class="col-12 col-md-4">
            {% image area.interest_icon width-100 %}
            <h2 class="h2-heading">{{ area.name }}</h2>
            <p class="...">{{ area.description }}</p>
            <a href="{{ area.url }}" class="...">{% trans "Learn more →" %}</a>
          </div>
        {% endwith %}
      {% endfor %}
```

File: network-api/networkapi/wagtailpages/utils.py [hunk 6]
```diff
@@ -206,3 +207,31 @@ def insert_panels_after(panels, after_label, additional_panels):
...
+    """
+
+    # The link here is handled as a dual field with custom validation, rather
+    # than a single field that lets users pick either an internal page or
+    # external URL, due to the diffuculties associated with doing that latter
+    # in a clean way. See https://github.com/mozilla/foundation.mozilla.org/issues/4936
+    # for a more detailed explanation
+
+    if model.internal_link and model.external_link:
+        # Both fields are filled out
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/utils.py:225]:
```suggestion
```

File: network-api/networkapi/wagtailpages/pagemodels/base.py (hunk 1 above)
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/pagemodels/base.py]: I'm going to mark this as resolved because we explored a way to do this already.

**Comment by KalobTaulien** [comment]
@Pomax this looks good to me. 👍

**Pull request merged**